cli/azd/pkg/project/service_target_springapp.go

```diff
@@ -0,0 +1,285 @@
+				task.SetError(fmt.Errorf("provisioning infrastructure for app deployment: %w", err))
+				return
+			}
+
+			if len(deployResult.Deployment.Outputs) > 0 {
+				log.Printf("saving %d deployment outputs", len(deployResult.Deployment.Outputs))
+				if err := provisioning.UpdateEnvironment(at.env, deployResult.Deployment.Outputs); err != nil {
+					task.SetError(fmt.Errorf("saving outputs to environment: %w", err))
+					return
+				}
+			}
+
+			if targetResource.ResourceName() == "" {
+				azureResource, err := at.resourceManager.GetServiceResource(
+					ctx,
+					targetResource.SubscriptionId(),
+					targetResource.ResourceGroupName(),
+					serviceConfig,
+					"deploy",
+				)
+				if err != nil {
+					task.SetError(err)
+					return
+				}
+
+				// Fill in the target resource
+				targetResource = environment.NewTargetResource(
+					targetResource.SubscriptionId(),
+					targetResource.ResourceGroupName(),
+					azureResource.Name,
+					azureResource.Type,
+				)
+
+				if err := checkResourceType(targetResource, infra.AzureResourceTypeSpringApp); err != nil {
+					task.SetError(err)
+					return
+				}
+			}
+
+			task.SetProgress(NewServiceProgress("Fetching endpoints for spring apps service"))
+			endpoints, err := at.Endpoints(ctx, serviceConfig, targetResource)
+			if err != nil {
+				task.SetError(err)
+				return
+			}
+
+			task.SetResult(&ServicePublishResult{
+				Package: packageOutput,
+				TargetResourceId: azure.SpringAppRID(
+					targetResource.SubscriptionId(),
+					targetResource.ResourceGroupName(),
+					targetResource.ResourceName(),
+				),
+				Kind:      SpringAppTarget,
+				Details:   deployResult,
+				Endpoints: endpoints,
+			})
+		},
+	)
+}
+
+// Gets the exposed endpoints for the Spring Apps Service
+func (st *springAppTarget) Endpoints(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	targetResource *environment.TargetResource,
+) ([]string, error) {
+	appServiceProperties, err := st.cli.GetSpringAppProperties(
+		ctx,
+		targetResource.SubscriptionId(),
+		targetResource.ResourceGroupName(),
+		targetResource.ResourceName(),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("fetching service properties: %w", err)
+	}
+
+	endpoints := make([]string, len(appServiceProperties.HostNames))
+	for idx, hostName := range appServiceProperties.HostNames {
+		endpoints[idx] = fmt.Sprintf("https://%s/", extractEndpoint(hostName))
+	}
+
+	return endpoints, nil
+}
+
+func (at *springAppTarget) validateTargetResource(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	targetResource *environment.TargetResource,
+) error {
+	if targetResource.ResourceGroupName() == "" {
+		return fmt.Errorf("missing resource group name: %s", targetResource.ResourceGroupName())
+	}
+
+	if targetResource.ResourceType() != "" {
+		if err := checkResourceType(targetResource, infra.AzureResourceTypeSpringApp); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func extractEndpoint(hostName string) string {
```

Comment:
Could you write a few tests for `extractEndpoint` in `service_target_springapp_test.go`?